src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -353,6 +353,16 @@ async def prepare_coursier_resolve_info(
         for (req, _), path in zip(jars, jar_file_paths)
     ]
 
+    # Coursier only fetches non-jar artifact types ("packaging" in Pants parlance) if passed an `-A` option
```

Comment:
Should this diff (and the corresponding change in `jvm_tool.py` be landed as a separate PR? It is necessary for Pants to actually get Coursier to download non-jar artifacts.

Reply:
Generally I think it'd be a good idea, but not a very big deal. This PR is small enough.

Reply:
Do you think this would be worth dedicated testing?

Reply:
Yes, I need to add a test. I'll break this out into its own PR.

Reply:
https://github.com/pantsbuild/pants/pull/16358